rainier-core/src/main/scala/com/stripe/rainier/compute/Decimal.scala

```diff
@@ -1,47 +1,129 @@
 package com.stripe.rainier.compute
 
-case class Decimal(value: BigDecimal) {
-  def toDouble = value.toDouble
-  def toInt = value.toInt
+import scala.annotation.tailrec
 
-  def isValidInt = value.isValidInt
-  def isWhole = value.isWhole
+sealed trait Decimal {
```

Comment:
odd, I would have thought that the `WartRemover` would make this comment - it should be running on compile.

If you use 

```
sealed trait Decimal extends Product with Serializable
```

type errors won't infer "Product with Serializable" as the type of case classes that extend this trait, and errors will be more nicely readable.

(Maybe that only applies to traits with generics?)